src/controller/OnOffCluster.cpp

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
In order to give you an idea of how this code will fit with templates, I have created an [example](https://github.com/vivien-apple/connectedhomeip-1/tree/ZAP_ExampleClusters)

This example will generate 2 files: `Clusters.h` and `Clusters.cpp` that contains the generated code for the wrappers of all clusters supported by the `all-clusters-app`

In order to generate the code, here are the steps you need to follow:
```
cd third_party/zap/repo
npm install
cd ../../../
./scripts/tools/zap_generate.sh examples/all-clusters-app/all-clusters-common/all-clusters-app.zap
```

Please note that I've not tried to compile the generated code, so it may contains some typos :)
Also, `ZAP` does not expose the `writable` field of attributes at the moment, so you will not see any `Write{{asCamelCased name false}}Attribute value generated.

In general you may want to use a single file that contains all the generated code, otherwise you will need to add a specific file per cluster in `src/app/zap-templates/chip-templates.json`. That's doable without having to duplicate the template code though. Depends on what folks prefers.

Also, I would say the generated `Clusters.h` and `Clusters.cpp` file belongs to `src/app` and not to `src/controller`.
The rationale here is that a device may implement the client side of a cluster, while `src/controller` is not targeting the device itself (at least that's what I think, please correct me if I'm wrong).
Now there may be some specific code depending if you run on a device or a controller, and that's the part that should belong to `src/controller` in your case.

Reply:
I tried running the script, and it looks really cool. Some tweaks will be needed, as the OnOffCluster.cpp has been updated, but it's so close now.
We can move the generated cluster to `src/app` in a separate PR. I suspect that'll require some other changes as well.

Reply:
I have opened https://github.com/project-chip/zap/pull/64 about exposing `writable` to the templates.